d4s/src/main/scala/d4s/DynamoClient.scala

```diff
@@ -17,9 +17,15 @@ object DynamoClient {
     dynamo: DynamoComponent
   ) extends DynamoClient[F] {
 
-    def raw[T](f: DynamoDbClient => T)(implicit notInMonadSanityCheck: T <:!< F[_, _]): F[Throwable, T] = {
+    def raw[T](f: DynamoDbClient => T)(implicit ev: NotMonad[T, F]): F[Throwable, T] = {
```

Comment:
@aliubymov-playq, IMO but previous variant was better to read and understand, it was like reading a simple condition.

Reply:
It depended on `shapeless.<!:<`, so we either have to copy it or write an alternative. but tbh this sanity check is not crucial, it's not required to keep it if it's in any way inconvenient

Reply:
@vpodlnk @neko-kai I've thought of deleting this but decided to left old behavior. So if you think that this part is not crucial, then I'll delete this check. (Anyway, I suppose that this check a bit useless)